framework/generated/base_generators/base_struct_decoders_header_generator.py

```diff
@@ -36,6 +35,11 @@ def generate_struct_info(self):
         """Performs C++ code generation for the feature."""
         body = '\n'
         for struct in self.all_structs:
+            if struct in self.all_struct_aliases or struct in self.all_unions or self.is_struct_black_listed(
```

Comment:
More DRY.  Have seen this multipart clause before should be a function, especially where it's new.